src/main/java/org/truffleruby/core/array/ArrayNodes.java

```diff
@@ -1850,6 +1851,9 @@ public int size(DynamicObject array,
 
         private final BranchProfile errorProfile = BranchProfile.create();
 
+        protected final Assumption fixnumCmpAssumption = Layouts.MODULE.getFields(getContext().getCoreLibrary().getFixnumClass()).registerAssumption("<=>");
+        protected final Assumption floatCmpAssumption = Layouts.MODULE.getFields(getContext().getCoreLibrary().getFloatClass()).registerAssumption("<=>");
```

Comment:
This is semantically incorrect, because it could end up registering an Assumption for a redefined `Fixnum#<=>`.
A SortNode can be created much later, due to splitting I think.
So you need to do it during context creation or initialization, and it would make sense to have them in `CoreMethods`.

Reply:
Both of these are defined in Java code so it should be no trouble to capture them when CoreMethods is created.

Reply:
Actually, `@CoreMethod` node are created lazily, see CoreMethodNodeManager.addLazyCoreMethod(), so the node will be class-loaded and created only on first usage.
Splitting might clone the node which might just copy these fields over without calling the constructor, but I wouldn't want to rely on that either.

Reply:
Do you mind if I change visibility of some operations in `CoreMethods` then? And I'll have to try to figure out why we were getting an NPE with the Substrate VM when I last tried that.

Reply:
I think you only need a couple getters (or public final fields) for the 2 Assumptions, no?
That NPE sounds worth figuring out in any case :)

Reply:
(Otherwise changing visibility is likely fine if there is a need)

Reply:
I got the NPE with public final fields. I'll look into that more.